wordpress-mobile/release-toolkit
# L10n: Add support for routines other than NSLocalisedStrings (for widgets)

Part of project ref paaHJt-2Ib-p2
Follow-up on #309

## What

This adds support to provide custom alternate routines in addition to the default `NSLocalizedString(…)` macros parsed by `genstrings`.

This will be quite useful in order to make Widgets look up their localized strings in the main app bundle instead of their own bundle.
Instead of having to pass `Bundle(identifier: the_app_bundle)` or similar in the widgets' code _every time_ we need an `NSLocalizedString` from the app in the widgets, we'll now be able to create a custom `func AppLocalizedString(…)` helper function in the widgets' codebase, and use that in their codebase, DRYing the widget code significantly.

## How

 - Added new `ConfigItem` which ends up adding the appropriate `-s <routine>` flags to the invocation of `genstrings`
 - Updated the unit tests accordingly

## To test

- Verify CI is green
- Review the `.strings` files I used as "expected" output for the unit tests and ensure they make sense according to the code being parsed.
